File: _inject_premium.py
"""Inject new premium v2 scripts into HTML files"""

files_scripts = {
    'editor.html':      '<script src="editor-premium-v2.js" defer></script>',
    'editor-en.html':   '<script src="editor-premium-v2.js" defer></script>',
    'viewer.html':      '<script src="viewer-premium-v2.js" defer></script>',
    'viewer-en.html':   '<script src="viewer-premium-v2.js" defer></script>',
}

for fname, tag in files_scripts.items():
    path = r'c:\Users\TheFo\Downloads\nfc\\' + fname
    try:
        with open(path, 'r', encoding='utf-8') as f:
            c = f.read()
        script_name = tag.split('"')[1]
        if script_name not in c:
            c = c.replace('</body>', tag + '\n</body>', 1)
            with open(path, 'w', encoding='utf-8') as f:
                f.write(c)
            print(f'OK: {fname}')
        else:
            print(f'SKIP (already exists): {fname}')
    except Exception as e:
        print(f'ERROR {fname}: {e}')

File: inject-premium.js
const fs = require('fs');
const path = require('path');

function injectPremiumUI() {
    const rootDir = __dirname;
    const files = fs.readdirSync(rootDir);
    const htmlFiles = files.filter(f => f.endsWith('.html'));

    const cssTag = '    <link rel="stylesheet" href="/nfc/premium-ui.css">\n';
    const jsTag = '    <script src="/nfc/premium-ui.js" defer></script>\n';

    let injectedCount = 0;

    for (const file of htmlFiles) {
        const filePath = path.join(rootDir, file);
        let content = fs.readFileSync(filePath, 'utf8');

        let modified = false;

        // Inject CSS right before </head>
        if (!content.includes('premium-ui.css') && content.includes('</head>')) {
            content = content.replace('</head>', cssTag + '</head>');
            modified = true;
        }

        // Inject JS right before </body>
        if (!content.includes('premium-ui.js') && content.includes('</body>')) {
            content = content.replace('</body>', jsTag + '</body>');
            modified = true;
        }

        if (modified) {
            fs.writeFileSync(filePath, content);
            console.log(`Injected into ${file}`);
            injectedCount++;
        }
    }

    console.log(`Successfully injected Premium UI into ${injectedCount} files.`);
}

injectPremiumUI();
